src/formGenerator.vue

```diff
@@ -124,11 +123,31 @@ export default {
 	methods: {
 		// Get visible prop of field
 		fieldVisible(field) {
-			if (isFunction(field.visible)) return field.visible.call(this, this.model, field, this);
+			let visible = field.visible;
+
+			if (isFunction(field.visible)) {
+				visible = field.visible.call(this, this.model, field, this);
+			}
+
+			if (isNil(field.visible)) {
+				visible = true;
+			}
 
-			if (isNil(field.visible)) return true;
+			// if the schema formOptions includes a deleteDataOnHide attribute:
+			// 1. we check if visibility became false on the field. If so, delete the model from that field
+			// 2. if became visible and we have an initial, we can set that initial back if the field not in model
+			if (this.options.deleteDataOnHide) {
+				if (!visible) {
+					this.$delete(this.model, field.model);
+				}
+				else if (visible && field.initial) {
+					if (!(field.model in this.model)) {
+						this.$set(this.model, field.model, field.initial);
```

Comment:
vue form generator allows for dotted paths to create complex option in model names, so `complex.model.name` will create nested objects. This is done via `setModelValueByPath` but it is essentially following the logic of `set` in lodash.

We have `vueSet` in our repo that maybe we want to add to the `utils` file in vfg.